docs/DeveloperGuide.md

```diff
@@ -2,14 +2,372 @@
+    
+   ##### Module Information Retrieval
+    
+   - **`getModuleInfo(String moduleCode)`**: Retrieves detailed information about a module, including its Modular Credits, description, and title, based on the module code.
+     - **Parameters**: `String moduleCode` - The code of the module for which information is to be retrieved.
+     - **Note**: This method updates the internal state of the `JsonManager` object with the retrieved module information.
+    
+   ##### Information Accessors
+    
+   - **`getModuleDescription()`**: Returns the description of the last module queried.
+     - **Returns**: `String` - The description of the module.
+   
+   - **`getModuleMC()`**: Returns the Modular Credits of the last module queried.
+     - **Returns**: `int` - The Modular Credits of the module.
+    
+   - **`getModuleTitle()`**: Returns the title of the last module queried.
+     - **Returns**: `String` - The title of the module.
+    
+   #### Error Handling
+    
+   - The constructor throws a `RuntimeException` if the JSON file containing module information cannot be found or accessed, ensuring that the application is aware of missing or inaccessible module data.
+    
+   #### Usage
+    
+   ```java
+   JsonManager jsonManager = new JsonManager();
+   if (jsonManager.moduleExist("CS1010")) {
+       jsonManager.getModuleInfo("CS1010");
+       System.out.println("Module Title: " + jsonManager.getModuleTitle());
+       System.out.println("Module Description: " + jsonManager.getModuleDescription());
+       System.out.println("Module MC: " + jsonManager.getModuleMC());
+   }
+   ```
+6. **Viewing modules left to graduate**
+
+   The `ViewGraduateCommand` class is responsible for displaying the list of modules that a student needs to
+   complete
+   for graduation with respect to the modules the user has previously completed. The diagram below provides an overview
+   of how this class interacts with other components in the
+   system.
+
+   ![View Graduate Module Class](diagrams/ViewGraduateModuleClass.png)
+
+   It utilizes the `Ui` class to print out the list of modules in a
+   formatted manner. Additionally, it interacts with the `CEGModules` enum to retrieve module information such as
+   module codes and Module Credits (MCs).
+7. **Viewing GPA**
+
+   The `ViewGpaCommand` class is responsible for displaying the current GPA attained by the student. It
+   accesses `ModuleList`, which looks through all `Module` object contained in the list. If the `Module` is marked as
+   taken and has been assigned a valid grade with `GradeCommand` by the user, its grades will be included into the
+   calculation.
+
+   This is the formula used for tabulation of GPA.
+   `GPA = SUM(Course Grade Point * Course Units) / SUM(Course Units Counted Towards GPA)`
+
+   Below is the sequence diagram for `ViewGpaCommand`.
+   ![View Gpa Command Sequence Diagram](diagrams/ViewGpaCommand.png)
```

Comment:
Overall good sequence diagram drawn.
For the ViewGpaCommand sequence diagram:
- Can write what output is returned to the user 
- Is the log message and GpaNullException supposed to be returned to ModuleList? (No need to write them for the return line)